test/integration/driver.go

```diff
@@ -432,33 +229,37 @@ func getRestartEpoch(pod string) (int, error) {
 	return 0, fmt.Errorf("could not obtain envoy restart epoch")
 }
 
-func addRule(ruleConfig []byte, kind string, name string, namespace string) error {
-
-	out, err := yaml.YAMLToJSON(ruleConfig)
-	if err != nil {
-		return fmt.Errorf("Cannot convert YAML rule to JSON: %v", err)
-	}
-
+func addConfig(config []byte, kind, name, namespace string) {
+	log.Println("Add config")
+	log.Println(string(config))
+	out, err := yaml.YAMLToJSON(config)
+	check(err)
 	istioKind, ok := model.IstioConfig[kind]
 	if !ok {
-		return fmt.Errorf("Invalid kind %s", kind)
+		check(fmt.Errorf("Invalid kind %s", kind))
 	}
 	v, err := istioKind.FromJSON(string(out))
-	if err != nil {
-		return fmt.Errorf("Cannot parse proto message from JSON: %v", err)
-	}
-
-	err = istioClient.Put(model.Key{
+	check(err)
+	check(istioClient.Put(model.Key{
 		Kind:      kind,
 		Name:      name,
 		Namespace: namespace,
-	}, v)
+	}, v))
+}
 
-	return err
+func deployConfig(config []byte, kind, name, namespace string, envoy string) {
+	epoch, err := getRestartEpoch(envoy)
+	check(err)
+	addConfig(config, kind, name, namespace)
+	check(waitForNewRestartEpoch(envoy, epoch))
 }
 
 func write(in string, data map[string]string, out io.Writer) error {
 	tmpl, err := template.ParseFiles(in)
+	data["namespace"] = namespace
+	data["hub"] = hub
+	data["tag"] = tag
+
```

Comment:
Shouldn't the 'err' check hug the template.parseFiles(), or do we need to unconditionally update the map regardless?